rs/src/gstd/calls.rs

```diff
@@ -46,47 +55,67 @@ impl GStdArgs {
     pub fn reply_deposit(&self) -> Option<GasUnit> {
         self.reply_deposit
     }
+
+    pub fn redirect_on_exit(&self) -> bool {
+        self.redirect_on_exit
+    }
 }
 
 #[derive(Debug, Default, Clone)]
 pub struct GStdRemoting;
 
 impl GStdRemoting {
-    fn send_for_reply(
+    pub fn new() -> Self {
+        Self
+    }
+
+    pub(crate) fn send_for_reply<T: AsRef<[u8]>>(
         target: ActorId,
-        payload: impl AsRef<[u8]>,
+        payload: T,
         #[cfg(not(feature = "ethexe"))] gas_limit: Option<GasUnit>,
         value: ValueUnit,
         #[allow(unused_variables)] args: GStdArgs,
-    ) -> Result<msg::MessageFuture, crate::errors::Error> {
+    ) -> Result<MessageFutureWithRedirect<T>> {
+        // here can be a redirect target
         #[cfg(not(feature = "ethexe"))]
```

Comment:
super hard to read the fn, maybe split it just for ethexe feat and not? 2 different impls